ui/app/mainui/Popups.qml

```diff
@@ -180,48 +201,56 @@ QtObject {
     }
 
     function openSendIDRequestPopup(publicKey, contactDetails, cb) {
-        openPopup(sendIDRequestPopupComponent, {
-            publicKey: publicKey,
-            contactDetails: contactDetails,
-            title: qsTr("Request ID verification"),
-            labelText: qsTr("Ask a question only they can answer"),
-            challengeText: qsTr("Ask your question..."),
-            buttonText: qsTr("Request ID verification")
-        }, cb)
+        const details = getContactDetails(publicKey)
+        details.title = qsTr("Request ID verification")
+        details.labelText = qsTr("Ask a question only they can answer")
+        details.challengeText = qsTr("Ask your question...")
+        details.buttonText = qsTr("Request ID verification")
+        openPopup(sendIDRequestPopupComponent, details, cb)
     }
 
-    function openMarkAsIDVerifiedPopup(publicKey, contactDetails, cb) {
-        openPopup(markAsIDVerifiedPopupComponent, {publicKey, contactDetails}, cb)
+    function openMarkAsIDVerifiedPopup(publicKey: string, contactDetails, cb) {
+        const details = getContactDetails(publicKey)
+        openPopup(markAsIDVerifiedPopupComponent, details, cb)
     }
 
-    function openRemoveIDVerificationDialog(publicKey, contactDetails, cb) {
-        openPopup(removeIDVerificationPopupComponent, {publicKey, contactDetails}, cb)
+    function openRemoveIDVerificationDialog(publicKey: string, contactDetails, cb) {
+        const details = getContactDetails(publicKey)
+        openPopup(removeIDVerificationPopupComponent, details, cb)
     }
 
-    function openOutgoingIDRequestPopup(publicKey, contactDetails, cb) {
-        let details = contactDetails ?? Utils.getContactDetailsAsJson(publicKey)
+    function openOutgoingIDRequestPopup(publicKey: string, contactDetails, cb) {
+        let details = getContactDetails(publicKey)
         try {
             const verificationDetails = rootStore.contactStore.getSentVerificationDetailsAsJson(publicKey)
```

Comment:
Ah we do have access to the contact store, so maybe we're fine with this update as an intermezzo